src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -80,6 +81,17 @@ public static IndexingConfig fromJson(JsonNode jsonNode) {
       }
       return new IndexingConfig(allowed, denied);
     }
+
+    // Need to override to prevent comparison of the supplier
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o instanceof IndexingConfig other) {
```

Comment:
I might be missing something but do we also need to check vector config here?

Reply:
This is within nested class, `IndeingConfig` and not for `CollectionSettings` -- `CollectionSettings` uses default, record-provided field-by-field equality check. So that will cover `vectorEnabled` and other fields.